pydantic/fields.py

```diff
@@ -689,22 +689,143 @@ class _EmptyKwargs(typing_extensions.TypedDict):
 }
 
 
+_T = TypeVar('_T')
+
+
+# NOTE: Actual return type is 'FieldInfo', but we want to help type checkers
```

Comment:
Maybe add a comment to each just so that it's clear, ex: `default set`, `default_factory set`, `no default set` or something along those lines

Reply:
Probably a stupid question - can we put these in an if type checking block for clarity?

Reply:
type checkers complain about the missing overload implementation iirc

Reply:
Gotcha.